### PR TITLE
SDIT-1220: ✨ Add in update details to smoke resource

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/SmokeTestHelperResource.kt [hunk 1]
```diff
@@ -23,6 +27,7 @@ import uk.gov.justice.hmpps.prison.service.SmokeTestHelperService
 @RequestMapping(value = ["\${api.base.path}/smoketest"], produces = ["application/json"])
 @Validated
 @ConditionalOnProperty(name = ["smoke.test.aware"], havingValue = "true")
+@PreAuthorize("hasRole('SMOKE_TEST') and hasAuthority('SCOPE_write')")
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/SmokeTestHelperResource.kt:30]: moved up to resource level

File: src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/repository/OffenderRepository.kt [hunk 2]
```diff
@@ -24,16 +22,12 @@ interface OffenderRepository : JpaRepository<Offender, Long> {
...
-    return findOffendersByNomsId(nomsId, PageRequest.of(0, 1)).stream().findFirst()
-  }
+  fun findOffenderByNomsId(nomsId: String): Optional<Offender> =
+    findOffendersByNomsId(nomsId, PageRequest.of(0, 1)).stream().findFirst()
 
   @Query("select o from Offender o left join fetch o.bookings b WHERE o.nomsId = :nomsId and b.bookingSequence = 1")
   fun findOffenderWithLatestBookingByNomsId(@Param("nomsId") nomsId: String): Optional<Offender>
 
-  @Query(value = "select o from Offender o join o.bookings ob join ob.images oi WHERE oi.captureDateTime > :start")
-  fun getOffendersWithImagesCapturedAfter(@Param("start")start: LocalDateTime, pageable: Pageable): Page<Offender>
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/repository/OffenderRepository.kt:35]: unused

File: src/main/java/uk/gov/justice/hmpps/prison/service/SmokeTestHelperService.kt [hunk 3]
```diff
@@ -18,10 +19,18 @@ class SmokeTestHelperService(
   private val offenderBookingRepository: OffenderBookingRepository,
   private val prisonerReleaseAndTransferService: PrisonerReleaseAndTransferService,
   private val bookingIntoPrisonService: BookingIntoPrisonService,
+  private val offenderRepository: OffenderRepository,
 ) {
   @Transactional
-  @VerifyBookingAccess(overrideRoles = ["SMOKE_TEST"])
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/service/SmokeTestHelperService.kt:23]: access restrictions moved to resource level

**Review by steverendell — APPROVED** [review]
lgtm

**Pull request merged**